cpp/src/arrow/compute/CMakeLists.txt

```diff
@@ -48,9 +48,16 @@ if(ARROW_TESTING AND ARROW_COMPUTE)
   add_library(arrow_compute_testing OBJECT ${ARROW_COMPUTE_TESTING_SRCS})
   # Even though this is still just an object library we still need to "link"
   # arrow_compute_core_testing so that is also included correctly
-  target_link_libraries(arrow_compute_testing
-                        PUBLIC $<TARGET_OBJECTS:arrow_compute_core_testing>
-                        PUBLIC ${ARROW_GTEST_GTEST_MAIN})
+  if(MSVC AND MSVC_VERSION LESS 1930) # gtest linkage needs compat on Visual Studio 2019
```

Comment:
Hmm. Is it really related to Visual Studio version?
It seems that this will be happened with newer Visual Studio.

Could you share the build log URL for this change?

Reply:
OK without the macro the SEH exceptions are gone, so maybe that was a temporal issue with AppVeyor. However, I still get the error about the invalid file path without this macro, which you can see in the latest run here:

https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/52266070

In the interim, I fixed up the macro (changed `_MSVC_VER` -> `_MSC_VER`) and it looks like things are all green, so hopefully that works for now